### PR TITLE
ML notebook

### PR DESCRIPTION
**Pull request by TolgoAI**
Addressed some of the machine-learning model feedback from Jan 29th, 2022

### CODE REVIEW

**Review by maximilian-hildebrandt — APPROVED** [review]
General:

- Pre-processing and data visualization already quite good with some smaller technical errors.

Exploratory data analysis:

- Missing values: "The are few missing values, therefore, it is statistics insignificant to drop all of them." I would suggest not to use the term statistically insignificant outside of the context of statistical inference (i.e., hypothesis testing & making inferences on populations), otherwise it is confusing. Better: "Due to low amount of missing values, it was decided to drop them."
- Frequency of cloudy weather appears too low. Is it really zero? Please check pre-processing/plots. 
- Correlation between temperature and feet items does not make sense, you have different classes of feet items, not an increasing interval-scale for feet. You can compute a correlation between temperature and heat perception for training purposes if you want. I would suggest you to read what the correlation coefficient actually represents and what requirements it has, since this is really important in practice. 
- Data visualization foundations are great, you created some good basic plots and got a better understanding of the data. However, interpretation is missing. What does this tell you? Interprete at least the box plot to form some hypotheses on the class rules.
- Outlook: A potential next step for data visualization in practice would be more fine-grained analyses: E.g., gender frequency for each feet class, heaviness of trip by feet class. In the end, you want to predict feet items, so visualize the data split by this class! (not necessary for this final project phase)
- Minor details:
- Order unique values by size to make it easier to read.
- Replacement of weather states: "feets['weather'].replace(['cloudy','sunny','snowy','rainy'],inplace=True)" Actually, you do this command a few lines later, then drop the first one? Without a second list of words, this command also does not generate any relevant output I think. 
- 2nd subchapter is empty. Rename subchapters without verbs & full sentences ("The third step is to create bar charts." => "Data Visualization").
- In 3rd sub-chapter you have pre-processing (sex categories replaced with numeric indicators). Move that to subchapter one in EDA.

Model training:

- Do not replace integers of classes with text strings. For model training, having numeric class labels is best practice.
- You performed a faulty train-test-validation split by splitting the same dataset into two overlapping sets. You don't want to use the same data twice ;) You need one train-test split using train_test_split(). Then you perform k-fold cross-validation on x_train, y_train. (Final evaluation on x_test, y_test after everything else has been optimized).
- Rest of workbook still WIP, will review once you refined this.
